tests/test_algorithms.py

```diff
@@ -169,13 +169,45 @@ def test_from_ts(self):
             assert ts.sequence_length == 100
 
     def test_pedigree(self):
-        tables = simulate_pedigree(num_founders=10, num_generations=5)
+        num_founders = 10
+        num_generations = 5
+        tables = simulate_pedigree(
+            num_founders=num_founders, num_generations=num_generations
+        )
         with tempfile.TemporaryDirectory() as tmpdir:
             ts_path = pathlib.Path(tmpdir) / "pedigree.trees"
             tables.dump(ts_path)
             ts = self.run_script(f"0 --from-ts {ts_path} -r 0.1 --model=wf_ped")
         assert ts.num_trees > 1
         for tree in ts.trees():
             assert tree.num_roots > 1
-        assert ts.num_individuals == len(tables.individuals)
-        assert ts.num_samples == 20
+        for node in ts.nodes():
+            assert node.individual != tskit.NULL
+        assert (
+            len(tables.individuals)
+            == ts.num_individuals
+            == len(ts.dump_tables().individuals)
+        )
+        assert len(ts.dump_tables().nodes) == len(tables.nodes)
+        assert ts.num_samples == 2 * num_founders
+
+        # founder nodes have correct time
+        founder_ids = {
+            idx
+            for idx, ind in enumerate(ts.dump_tables().individuals)
+            if np.array_equal(ind.parents, [-1, -1])
+        }
+        assert all(
+            [
+                (node.individual in founder_ids)
+                for node in ts.dump_tables().nodes
+                if node.time == (num_generations - 1)
+            ]
+        )
+
```

Comment:
So we're not really testing the trees here. What about something like
```python
for tree in ts.trees():
    for node_id in tree.nodes():
         node = ts.node(node_id)
         individual = ts.individual(node.individual)
         if tree.parent(node_id) != tskit.NULL:
              parent_node = ts.node(tree.parent(node_id))
              assert parent_node.individual in individual.parents
         else:
              assert node.individual in founder_ids
```
right? This checks that the trees agree with the pedigree.

Reply:
This is throwing an error because I'm getting trees that look like this: 
```
             36      35         21                 31
             ┏┻━┓   ┏━┻━━┓    ┏━━┻━━┓            ┏━━┻━┓
     55      ┃  ┃  48    ┃   46     ┃      52    ┃    ┃
    ┏━┻━┓    ┃  ┃  ┏┻━┓  ┃  ┏━┻┓    ┃      ┏┻━┓  ┃    ┃
    ┃  69    ┃  ┃  ┃  ┃  ┃  ┃  ┃   76      ┃  ┃  ┃   79
    ┃  ┏┻━┓  ┃  ┃  ┃  ┃  ┃  ┃  ┃  ┏━┻┓     ┃  ┃  ┃  ┏━┻┓
80 81 92 94 82 83 84 90 93 85 91 96 98 86 87 89 88 97 99 95
```
Would we expect the nodes of founders (0-19) to be missing?

Reply:
Hmm, so it looks like we're missing edges joining the nodes where coalescences occured to the founders. To have a full record, we'd need to have "unary" edges joining (say) 80 back to its founder.

I'll take a look, this is an important detail.